components/vfs/archive.hpp

```diff
@@ -21,6 +21,9 @@ namespace VFS
     public:
         virtual ~Archive() {}
 
+        /// Clears cached data for archives that may change.
```

Comment:
I avoided it because I wasn't sure how expensive indexing BSA's were, and I thought creating BSA's was one of the things modders do at the end when they are packaging their mods. If it's cheap though, then BSA's might as well be indexed.